### PR TITLE
[MIG] sale_order_action_invoice_create_hook: Migration to 12.0

### PR DESCRIPTION
**Pull request by rven**

### CODE REVIEW

File: sale_order_action_invoice_create_hook/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+{
+    "name": "Sale Order Action Invoice Create Hook",
+    "summary": "Add more flexibility in the grouping parameters "
+               "for the creation of invoices",
+    "author": "Eficent, Odoo Community Association (OCA)",
+    "version": "12.0.1.0.0",
+    "category": "Sale Workflow",
+    "website": "https://github.com/OCA/sale-workflow",
+    "license": 'AGPL-3',
```
Comment (nikul-serpentcs) [inline sale_order_action_invoice_create_hook/__manifest__.py]: @rven LGPL-3

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improve Code

**Review by jimhoefnagels — APPROVED** [review]

**Comment by tbaden** [comment]
@nikul-serpentcs changes were  done. anything else or good to go?

**Review by nikul-serpentcs — APPROVED** [review]
Only Code review Done from my side
 LGTM :+1:

File: sale_order_action_invoice_create_hook/hooks.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                'mail.message_origin_link',
+                values={'self': invoice, 'origin': references[invoice]},
+                subtype_id=self.env.ref('mail.mt_note').id)
+        return [inv.id for inv in invoices.values()]
+
+    if not hasattr(SaleOrder, 'action_invoice_create_original'):
+        SaleOrder.action_invoice_create_original = \
+            SaleOrder.action_invoice_create
+
+    SaleOrder.action_invoice_create = new_action_invoice_create
```
Comment (pedrobaeza) [inline sale_order_action_invoice_create_hook/hooks.py]: This should be done in `_register_hook` method and not here.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_order_action_invoice_create_hook/hooks.py (hunk 2 above)
Comment (rven) [reply sale_order_action_invoice_create_hook/hooks.py]: Ok like this?

File: sale_order_action_invoice_create_hook/model/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                    'mail.message_origin_link',
+                    values={'self': invoice, 'origin': references[invoice]},
+                    subtype_id=self.env.ref('mail.mt_note').id)
+            return [inv.id for inv in invoices.values()]
+
+        if not hasattr(Sale, 'action_invoice_create_original'):
+            Sale.action_invoice_create_original = \
+                Sale.action_invoice_create
+
+        Sale.action_invoice_create = new_action_invoice_create
```
Comment (pedrobaeza) [inline sale_order_action_invoice_create_hook/model/sale_order.py]: This is still monkey-patching all possible DBs in the same server. You have to use the method `_patch_method`:
```python
self._patch_method('action_invoice_create', new_action_invoice_create)
```

Comment (rven) [reply sale_order_action_invoice_create_hook/model/sale_order.py]: Is this ok now?

**Comment by pedrobaeza** [comment]
Almost there... please fix this:
```
sale_order_action_invoice_create_hook/model/sale_order.py:7:1: F401 'odoo.addons.sale.models.sale.SaleOrder as Sale' imported but unused
```

**Comment by rven** [comment]
Done!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**